indra/newview/llviewerparcelmgr.cpp

```diff
@@ -2717,16 +2721,32 @@ void LLViewerParcelMgr::onTeleportFinished(bool local, const LLVector3d& new_pos
         // Let's wait for the update and then emit the signal.
         mTeleportInProgressPosition = new_pos;
         mTeleportInProgress = true;
+        mTeleportWithinRegion = local;
     }
 }
 
 void LLViewerParcelMgr::onTeleportFailed()
 {
     mTeleportFailedSignal();
+    LLEventPumps::instance().obtain("LLTeleport").post(llsd::map("success", false));
 }
 
 bool  LLViewerParcelMgr::getTeleportInProgress()
 {
     return mTeleportInProgress // case where parcel data arrives after teleport
         || gAgent.getTeleportState() > LLAgent::TELEPORT_NONE; // For LOCAL, no mTeleportInProgress
 }
+
+void LLViewerParcelMgr::postTeleportFinished(bool local)
+{
+    auto post = []() { LLEventPumps::instance().obtain("LLTeleport").post(llsd::map("success", true)); };
+    if (local)
+    {
+        static LLCachedControl<F32> teleport_local_delay(gSavedSettings, "TeleportLocalDelay");
+        LL::Timers::instance().scheduleAfter(post, teleport_local_delay + 0.5f);
```

Comment:
`getInstance()->mTeleportFinishedSignal(new_pos, local);` is fired immediately for teleports within a region/sim. But actual teleport occurs after that delay. According to the comment in the code it's needed because "progress screen active but not shown".
But I agree that we may need to update this teleportation code at some point,